envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ControlPlane.kt

```diff
@@ -85,39 +87,29 @@ class ControlPlane private constructor(
         val properties: EnvoyControlProperties,
         val meterRegistry: MeterRegistry
     ) {
-        var grpcServerExecutor: Executor? = null
-        var nioEventLoopExecutor: Executor? = null
-        var executorGroup: ExecutorGroup? = null
-        var globalSnapshotExecutor: Executor? = null
+        var grpcServerExecutor: Executor = buildThreadPoolExecutor()
+        // unbounded executor - netty will only use configured number of threads
+        // (by nioEventLoopThreadCount property or default netty value: <number of CPUs> * 2)
+        var nioEventLoopExecutor: Executor = newMeteredCachedThreadPool("grpc-worker-event-loop")
+        var executorGroup: ExecutorGroup = buildExecutorGroup()
+        var globalSnapshotExecutor: Executor = newMeteredFixedThreadPool(
+            "snapshot-update",
+            properties.server.globalSnapshotUpdatePoolSize
+        )
+        var globalSnapshotAuditExecutor: Executor = newMeteredFixedThreadPool(
```

Comment:
This is valid approach for our usage, but if sb use method withGlobalSnapshotAuditExecutor our builder will create additional ThreadPool. 

I prefer lazy approach